lib/WordDiff.ml

```diff
@@ -31,131 +29,189 @@ let diff_words (s1 : string) (s2 : string) : line_content * line_content =
     | x :: xs, y :: ys, z :: zs -> (
         match (x = z, y = z) with
         | true, true ->
-            construct_diff xs ys zs (Unchanged x :: acc_mine)
-              (Unchanged y :: acc_their)
+            construct_diff xs ys zs
+              (Types.Unchanged x :: acc_mine)
+              (Types.Unchanged y :: acc_their)
         | false, true ->
-            construct_diff xs (y :: ys) (z :: zs) (Changed x :: acc_mine)
+            construct_diff xs (y :: ys) (z :: zs)
+              (Types.Changed x :: acc_mine)
               acc_their
         | true, false ->
             construct_diff (x :: xs) ys (z :: zs) acc_mine
-              (Changed y :: acc_their)
+              (Types.Changed y :: acc_their)
         | false, false ->
-            construct_diff xs ys (z :: zs) (Changed x :: acc_mine)
-              (Changed y :: acc_their))
+            construct_diff xs ys (z :: zs)
+              (Types.Changed x :: acc_mine)
+              (Types.Changed y :: acc_their))
     | x :: xs, [], lcs ->
-        construct_diff xs [] lcs (Changed x :: acc_mine) acc_their
+        construct_diff xs [] lcs (Types.Changed x :: acc_mine) acc_their
     | [], y :: ys, lcs ->
-        construct_diff [] ys lcs acc_mine (Changed y :: acc_their)
+        construct_diff [] ys lcs acc_mine (Types.Changed y :: acc_their)
     | x :: xs, y :: ys, [] ->
-        construct_diff xs ys [] (Changed x :: acc_mine) (Changed y :: acc_their)
+        construct_diff xs ys []
+          (Types.Changed x :: acc_mine)
+          (Types.Changed y :: acc_their)
     | [], [], _ :: _ -> assert false
     (* Since lcs is the longest common subsequence, this case cannot happen *)
   in
 
   construct_diff words1 words2 common [] []
 
-let compute (block : string Block.t) : line_content Block.t =
+let compute (block : string Block.t) : Types.line_content Block.t =
   match block with
-  | Block.Common line -> Block.Common [ Unchanged line ]
+  | Block.Common line -> Block.Common [ Types.Unchanged line ]
   | Block.Changed { mine; their; order } ->
       let mine_str = String.concat " " mine in
       let their_str = String.concat " " their in
       let mine_words, their_words = diff_words mine_str their_str in
       Block.Changed { mine = [ mine_words ]; their = [ their_words ]; order }
 
-let word_to_ui word attr = W.string ~attr (word ^ " ")
+let word_to_ui word attr rendering_mode diff_type =
+  match rendering_mode with
+  | Types.Color -> W.string ~attr (word ^ " ")
+  | Types.TextMarkers -> (
+      match diff_type with
+      | `Equal -> W.string (word ^ " ")
+      | `Deleted -> W.string ("<-" ^ word ^ "/-> ")
+      | `Added -> W.string ("<+" ^ word ^ "/+> "))
```

Comment:
Refactor into a market variable.

Reply:
In another PR it might be good to consider "grouping" consecutive added/removed words using single annotations, especially when using "opening-and-closing" markers.